qemu/tests/numa_stress.py

```diff
@@ -68,57 +71,109 @@ def run(test, params, env):
+        process.system(start_numad_service)
+
     timeout = float(params.get("login_timeout", 240))
+    stress_time = int(params.get("stress_time", 60))
     test_count = int(params.get("test_count", 4))
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
+    maxcpus = vm.cpuinfo.maxcpus
     session = vm.wait_for_login(timeout=timeout)
 
     qemu_pid = vm.get_pid()
 
     if test_count < len(host_numa_node.online_nodes):
         test_count = len(host_numa_node.online_nodes)
 
-    tmpfs_size = 0
-    for node in host_numa_node.nodes:
-        node_mem = int(host_numa_node.read_from_node_meminfo(node, "MemTotal"))
-        if tmpfs_size < node_mem:
-            tmpfs_size = node_mem
     tmpfs_path = params.get("tmpfs_path", "tmpfs_numa_test")
     tmpfs_path = utils_misc.get_path(data_dir.get_tmp_dir(), tmpfs_path)
     tmpfs_write_speed = get_tmpfs_write_speed()
-    dd_timeout = tmpfs_size / tmpfs_write_speed * 1.5
-    mount_fs_size = "size=%dK" % tmpfs_size
     memory_file = utils_misc.get_path(tmpfs_path, "test")
-    dd_cmd = "dd if=/dev/urandom of=%s bs=1k count=%s" % (memory_file,
-                                                          tmpfs_size)
     utils_memory.drop_caches()
 
-    if utils_memory.freememtotal() < tmpfs_size:
-        test.cancel("Host does not have enough free memory to run the test, "
-                    "skipping test...")
-
     if not os.path.isdir(tmpfs_path):
         os.mkdir(tmpfs_path)
 
     test_mem = float(params.get("mem"))*float(params.get("mem_ratio", 0.8))
-    stress_args = "--cpu 4 --io 4 --vm 2 --vm-bytes %sM" % int(test_mem / 2)
     most_used_node, memory_used = max_mem_map_node(host_numa_node, qemu_pid)
+    os_type = params["os_type"]
+
+    if os_type == "windows":
+        install_path = params["install_path"]
+        heavyload_install()
+    else:
+        stress_args = "--cpu %s --vm-bytes %sM --timeout %s" % (maxcpus,
+                                                                test_mem,
+                                                                stress_time)
+        stress_test = utils_test.VMStress(vm, "stress", params,
+                                          stress_args=stress_args)
+        stress_test.install()
 
     for test_round in range(test_count):
         if os.path.exists(memory_file):
             os.remove(memory_file)
         utils_memory.drop_caches()
-        if utils_memory.freememtotal() < tmpfs_size:
-            test.error("Don't have enough memory to execute this "
-                       "test after %s round" % test_round)
         error_context.context("Executing stress test round: %s" % test_round,
                               logging.info)
         numa_node_malloc = most_used_node
+        tmpfs_size = int(host_numa_node.read_from_node_meminfo(numa_node_malloc,
+                                                               "MemFree"))
+        if utils_memory.freememtotal() - tmpfs_size < 1024*1024:
```

Comment:
I don't think so. I set this value because sometimes other nodes only have hundreds of MB free memory, as every node is short of free memory, numad wouldn't balance qemu to other nodes.

Reply:
Got it, thanks